chain-signatures/node/src/protocol/message/mod.rs

```diff
@@ -361,20 +331,38 @@ impl MessageChannel {
             tracing::error!(?err, "outbox: failed to send message to participants");
         }
     }
+
+    /// Marks this message as filtered. This is used to prevent the same message with the
+    /// corresponding MessageId from being processed again.
+    pub async fn filter<M: MessageFilterId>(&self, msg: &M) {
+        if let Err(err) = self.filter.send((M::PROTOCOL, msg.id())).await {
+            tracing::warn!(?err, "failed to send filter message");
+        }
+    }
+
+    pub async fn filter_triple(&self, id: TripleId) {
+        if let Err(err) = self.filter.send((Protocols::Triple, id)).await {
+            tracing::warn!(?err, "failed to send filter message");
+        }
+    }
+
+    pub async fn filter_presignature(&self, id: PresignatureId) {
+        if let Err(err) = self.filter.send((Protocols::Presignature, id)).await {
+            tracing::warn!(?err, "failed to send filter message");
+        }
+    }
+
+    pub async fn filter_sign(&self, sign_id: SignId, presign_id: PresignatureId) {
+        self.filter(&(sign_id, presign_id)).await;
+    }
```

Comment:
Shouldn't it work like this? Seems like you built an abstraction without actually using it.

```suggestion
    /// Marks this message as filtered. This is used to prevent the same message with the
    /// corresponding MessageId from being processed again.
    async fn filter<M: MessageFilterId>(&self, msg: &M) {
        if let Err(err) = self.filter.send((M::PROTOCOL, msg.id())).await {
            tracing::warn!(?err, "failed to send filter message");
        }
    }

    pub async fn filter_triple(&self, id: TripleId) {
        self.filter(&id).await;
    }

    pub async fn filter_presignature(&self, id: PresignatureId) {
        self.filter(&id).await;
    }

    pub async fn filter_sign(&self, sign_id: SignId, presign_id: PresignatureId) {
        self.filter(&(sign_id, presign_id)).await;
    }
```